### PR TITLE
Make next steps page for 2122 PDF

### PR DESCRIPTION
**Pull request by opticbob**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *This work is behind a feature toggle (flipper): YES*
- *(Summarize the changes that have been made to the platform):* The next steps PDF page that was on the 2122a PDF is now on the 2122 as well.  There was also a lot of standardization of test data to make future generation of the fixture PDFs easier.
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution?)*
- *(Which team do you work for, does your team own the maintenance of this component?):* `FAR/ARM`
- *(If introducing a flipper, what is the success criteria being targeted?)*

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/88987
- *Link to previous change of the code/bug (if applicable)*
- *Link to epic if not included in ticket*

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change:* The generated 2122 had no next steps page, now it does.
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing:* Run through the appoint a rep workflow choosing an individual representative instead of a VSO and request a PDF at the end.
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
Screenshot 2024-10-23 105200

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas):* This only touches the `representation_management` module.

## Acceptance criteria

- [x]  I updated unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/representation_management/app/models/representation_management/form_2122_base.rb [hunk 1]
```diff
@@ -95,6 +100,12 @@ class Form2122Base
       validates :claimant_phone, length: { is: 10 }, format: { with: TEN_DIGIT_NUMBER }
     end
 
+    validates :representative_id, presence: true
+
+    def representative
```
Comment (holdenhinkle) [inline modules/representation_management/app/models/representation_management/form_2122_base.rb:105]: I don't think `#representative` is being called anywhere, is it? I couldn't find it being called. I pulled the branch, commented it out, ran the tests, and they all passed.

File: modules/representation_management/app/models/representation_management/form_2122_base.rb [hunk 2]
```diff
@@ -107,5 +118,16 @@ def consent_limits_must_contain_valid_values
         end
       end
     end
+
+    def find_representative
+      AccreditedIndividual.find_by(id: representative_id) ||
+        Veteran::Service::Representative.find_by(representative_id:)
+    end
+
+    def representative_resolves
```
Comment (holdenhinkle) [inline modules/representation_management/app/models/representation_management/form_2122_base.rb:127]: Should `return unless find_representative.nil?` be `return unless representative.nil?`? Make use of `#find_representative` and the memoization?

**Review by holdenhinkle — COMMENTED** [review]
I started the review. The kids have a late-start-day at school; I have to help them get ready and drive them. I'll finish up the review when I get back.

**Comment by holdenhinkle** [comment]
@opticbob Can you please add a screenshot to the PR description of the PDF page this generates?

File: modules/representation_management/app/models/representation_management/form_2122_base.rb (hunk 1 above)
Comment (opticbob) [reply modules/representation_management/app/models/representation_management/form_2122_base.rb:105]: It is used to lookup the representative values for `modules/representation_management/lib/representation_management/v0/pdf_constructor/form_2122.rb` and `modules/representation_management/lib/representation_management/v0/pdf_constructor/form_2122a.rb`.  It's used in a few tests as well.  If it's commented out `rspec modules/representation_management/spec` should get pretty ugly.

File: modules/representation_management/app/models/representation_management/form_2122_base.rb (hunk 2 above)
Comment (opticbob) [reply modules/representation_management/app/models/representation_management/form_2122_base.rb:127]: That makes sense!  I made the change here and for `organization_name` in the 2122 as well.

**Comment by opticbob** [comment]
@holdenhinkle The screenshot is up.

**Review by holdenhinkle — COMMENTED** [review]
Slowing moving through this. Just hopped into another meeting.

File: modules/representation_management/app/models/representation_management/form_2122_base.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (opticbob) [reply modules/representation_management/app/models/representation_management/form_2122_base.rb:127]: Sounds fine to me, I'll make the change.

[the remark this answers is not in the document]
Comment (opticbob) [reply modules/representation_management/app/models/representation_management/form_2122_base.rb]: Done!

File: modules/representation_management/app/models/representation_management/form_2122_data.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@ def find_organization_name
     end
 
     def organization_name_resolves?
```
Comment (holdenhinkle) [inline modules/representation_management/app/models/representation_management/form_2122_data.rb:22]: Can you please make this `#organization_exists?` to match `representative_exists?`?

**Review by holdenhinkle — COMMENTED** [review]
LGTM. One last thing.

File: modules/representation_management/app/models/representation_management/form_2122_data.rb (hunk 3 above)
Comment (opticbob) [reply modules/representation_management/app/models/representation_management/form_2122_data.rb]: Done!

**Review by holdenhinkle — APPROVED** [review]

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**